examples/pytorch/gin/model.py

```diff
@@ -0,0 +1,258 @@
+            # print(str(id(h)))
+            for layer in range(self.num_layers - 1):
+                # get_logger().info("h shape: " + str(self.h.shape))
+                # get_logger().info("linear layer shape: " + str(self.linears[layer]))
+                self.h = F.relu(self.batch_norms[layer](self.linears[layer](self.h)))
+            return self.linears[self.num_layers - 1](self.h)
+
+
+class GIN(BaseSupervisedModel):
+    """Simple supervised GIN model."""
+
+    def __init__(
+        self, 
+        num_layers: int, 
+        num_mlp_layers: int, 
+        edge_type: np.array,
+        input_dim:int,
+        label_idx: int,
+        label_dim: int,
+        hidden_dim: int, 
+        output_dim: int,
+        feature_type: FeatureType,
+        feature_idx: int,
+        feature_dim: int, 
+        final_dropout: str, 
+        learn_eps: bool, 
+        graph_pooling_type: str,
+        neighbor_pooling_type: str, 
+        device: None,
+        metric: BaseMetric = MRR(),
+        feature_enc: Optional[FeatureEncoder] = None,
+    ):
+
+        super(GIN, self).__init__(
+            feature_type=feature_type,
+            feature_idx=feature_idx,
+            feature_dim=feature_dim,
+            feature_enc=feature_enc,
+        )
+
+        self.final_dropout = final_dropout
+        self.device = device
+        self.metric = metric
+        self.edge_type = edge_type
+        self.num_layers = num_layers
+        self.num_mlp_layers = num_mlp_layers
+        self.graph_pooling_type = graph_pooling_type
+        self.neighbor_pooling_type = neighbor_pooling_type
+        self.learn_eps = learn_eps
+        self.eps = nn.Parameter(torch.zeros(self.num_layers-1))
+
+        self.label_idx = label_idx
+        self.label_dim = label_dim
+        self.feature_type = feature_type
+        self.feature_idx = feature_idx
+        self.feature_dim = feature_dim
+        self.hidden_dim = hidden_dim
+        self.output_dim = output_dim
+
+        self.mlps = torch.nn.ModuleList()
+
+        # Batch Norms to be applied on final layer
+        self.batch_norms = torch.nn.ModuleList()
+
+        for layer in range(self.num_layers-1):
+            if layer == 0:
+                self.mlps.append(MLP(num_mlp_layers, input_dim, hidden_dim, hidden_dim))
+            else:
+                self.mlps.append(MLP(num_mlp_layers, hidden_dim, hidden_dim, hidden_dim))
+
+            self.batch_norms.append(nn.BatchNorm1d(hidden_dim))
+
+        # Linear Function: hidden rep -> prediction score
+        self.linears_prediction = torch.nn.ModuleList()
+        for layer in range(num_layers):
+            if layer == 0:
+                self.linears_prediction.append(nn.Linear(input_dim, output_dim))
+            else:
+                self.linears_prediction.append(nn.Linear(hidden_dim, output_dim))
+    
+    def next_layer(self, h, layer, features, nb_counts, num_nodes):
+        pooled_rep = self.mlps[layer](h)
```

Comment:
Looks like H is not pooled here like in the example, this is probably important for learning